src/main/java/xyz/nucleoid/dungeons/dungeons/game/loot/weapon/DgMetalMeleeWeapon.java

```diff
@@ -112,12 +131,14 @@ private static String generateFlavourText(Random random, DgLootGrade grade, DgWe
 
     public ItemStack toItemStack() {
         ItemStackBuilder builder = ItemStackBuilder.of(this.type.asVanillaItem())
-                .setName(new LiteralText(String.format("%s %s %s", this.grade.name, this.metal.name, this.type.name)))
+                .setName(new TranslatableText("item.dungeons." + this.type.id, new TranslatableText("grade." + this.grade.id), new TranslatableText("material.metal." + this.metal.id)))
```

Comment:
- This is wrong for english because it yields "Battleaxe Battered Copper", does it not?
- I'm not sure this will work for other languages. They may not all have the same word order here.

Reply:
Nevermind me, this works fine